react/components/EXPERIMENTAL_FilterBar/FilterTag.js

```diff
@@ -0,0 +1,282 @@
+/* eslint-disable */
+import React, { PureComponent, Fragment } from 'react'
+import PropTypes from 'prop-types'
+import merge from 'lodash/merge'
+
+import IconClear from '../icon/Clear'
+import IconClose from '../icon/Close'
+import IconCaretDown from '../icon/CaretDown'
+import Statement from './UncoupledStatement'
+import Menu from './Menu'
+
+const OPEN_MENU_STYLE = { backgroundColor: '#dbe9fd' }
+const emptyVirtualStatement = {
+  subject: null,
+  verb: null,
+  object: null,
+  error: null,
+}
+
+const filterStatementBySubject = (statements = [], sbj, options = {}) => {
+  const hasStatement = statements.some(st => st.subject === sbj)
+  return hasStatement
+    ? statements.filter(st => st.subject === sbj)
+    : [
+        {
+          ...emptyVirtualStatement,
+          subject: sbj,
+          verb:
+            options[sbj] && options[sbj].verbs.length === 1
+              ? options[sbj].verbs[0].value
+              : null,
+        },
+      ]
+}
+
+class FilterTag extends PureComponent {
+  constructor(props) {
+    super(props)
+    this.filterMenuContainer = React.createRef()
+
+    this.state = {
+      isMenuOpen: false,
+      virtualStatement: filterStatementBySubject(
+        props.statements,
+        props.subject,
+        props.options
+      )[0],
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
I think not returning anything when the if isn't met throws here... I'd double-check it, but maybe return `prevState` at the end would do the trick :)

Reply:
Por baixo dos panos é o que o react faz se não me engano:
```
if (partialState === null || partialState === undefined) {
        // Null and undefined are treated as no-ops.
        return prevState;
      }
```

source: https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactUpdateQueue.js

Reply:
Ok. It's because I've had this error thrown on a component I made `Warning: Component.getDerivedStateFromProps():A valid state object (or null) must be returned. You have returned undefined.` 😓 

But if you're sure that React internals are accepting this now, no problem 👍